aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java

```diff
@@ -279,9 +288,24 @@ public void onResult(SignUpResult result) {
 
                 @Override
                 public void onError(Exception error) {
-                    onException.accept(
-                        new AuthException("Sign up failed", error, "See attached exception for more details")
-                    );
+                    if (error instanceof UsernameExistsException) {
+                        onException.accept(
+                                new AuthException.AWSCognitoAuthException.UsernameExistsException(error.getCause())
```

Comment:
Hey, I think as we talked about on Slack, errors like these (UsernameExists, NetworkException, etc.) aren't specific to Cognito so it'd be better just to have them up at the Global level of AuthException instead of making the subclass of AWSCognitoAuthException for them